Lombiq.Tests.UI/Helpers/ReliabilityHelper.cs

```diff
@@ -198,5 +250,19 @@ public static void DoWithRetriesOrFail(Func<bool> process, TimeSpan? timeout = n
 
             return (wait.Until(_ => process()), wait);
         }
+
+        private static async Task<(bool IsSuccess, SafeWait<object> Wait)> DoWithRetriesInternalAsync(
+            Func<Task<bool>> processAsync,
+            TimeSpan? timeout = null,
+            TimeSpan? interval = null)
+        {
+            var wait = new SafeWait<object>(new object());
+
+            // If no values are supplied then the defaults specified in AtataFactory will be used.
+            if (timeout != null) wait.Timeout = timeout.Value;
+            if (interval != null) wait.PollingInterval = interval.Value;
+
+            return (await wait.Until(async _ => await processAsync()), wait);
```

Comment:
`Until()` doesn't actually support async, it'll treat the `Task` return value as any other value.

Reply:
Ahh, but this is somewhat different, bcs the SafeWait.Until repeatedly does the process until it times out or if it runs OK.
So here we should implement an async version of it, not sure if that is already done somewhere else as it doesn't seem to be trivial.

Reply:
Check if anybody has done this. Otherwise, it seems pretty simple, just change the method to be async, the parameter to `Func<T, Task<TResult>> condition`, and add an await here:
![image](https://user-images.githubusercontent.com/1976647/155175981-2ea5ac44-3f2d-4c51-ac41-9cfb3ecc4416.png)